deploy/0_deploy.ts

```diff
@@ -41,82 +62,176 @@ const ensPublicResolver: PublicResolver = hre.artifacts.require(
  * features available in it.
  */
 module.exports = async function () {
-    console.log('Deploying PoCo Nominal..');
-    const accounts = await hre.web3.eth.getAccounts();
-    await initial_migration();
-    await deploy_token(accounts);
-    await deploy_core(accounts);
-    await deploy_ens(accounts);
-    // Retrieve proxy address from previous truffle-fixture deployment
-    const { address: erc1538ProxyAddress } = await erc1538Proxy.deployed();
-    if (!erc1538ProxyAddress) {
-        console.error('Failed to retrieve deployed address of ERC1538Proxy');
-        process.exitCode = 1;
-    }
-    console.log(`ERC1538Proxy found: ${erc1538ProxyAddress}`);
-    // Save addresses of deployed PoCo Nominal contracts for later use
-    saveDeployedAddress('ERC1538Proxy', erc1538ProxyAddress);
-
-    // Save addresses of deployed ENS contracts for later use
-    const { address: ensRegistryAddress } = await ensRegistry.deployed();
-    saveDeployedAddress('ENSRegistry', ensRegistryAddress);
-    const { address: ensPublicResolverAddress } = await ensPublicResolver.deployed();
-    saveDeployedAddress('ENSPublicResolver', ensPublicResolverAddress);
-
-    console.log('Deploying PoCo Boost..');
+    console.log('Deploying PoCo..');
+    // Deploy GenericFactory (if not already done)
+    const chainId = (await ethers.provider.getNetwork()).chainId;
     const [owner] = await hre.ethers.getSigners();
-    const iexecPocoBoostDeployment = await hre.deployments.deploy('IexecPocoBoostDelegate', {
-        libraries: {
-            IexecLibOrders_v5: (await IexecLibOrders.deployed()).address,
-        },
-        from: owner.address,
-        log: true,
-    });
-    console.log(`IexecPocoBoostDelegate deployed: ${iexecPocoBoostDeployment.address}`);
-    const IexecPocoBoostAccessorsDeployment = await hre.deployments.deploy(
-        'IexecPocoBoostAccessorsDelegate',
-        {
-            from: owner.address,
-            log: true,
-        },
-    );
-    console.log(
-        `IexecPocoBoostAccessorsDelegate deployed: ${IexecPocoBoostAccessorsDeployment.address}`,
+    const factoryDeployer = new Deployer(owner);
+    await factoryDeployer.ready();
+    genericFactoryInstance = GenericFactory__factory.connect(FACTORY.address, owner);
+    // Deploy RLC
+    const deploymentOptions = CONFIG.chains[chainId] || CONFIG.chains.default;
+    salt = process.env.SALT || deploymentOptions.v5.salt || ethers.constants.HashZero;
+    const isTokenMode = deploymentOptions.asset == 'Token';
+    let rlcInstanceAddress = isTokenMode
+        ? deploymentOptions.token // token
+            ? deploymentOptions.token
+            : await new RLC__factory()
+                  .connect(owner)
+                  .deploy()
+                  .then((contract) => {
+                      contract.deployed();
+                      return contract.address;
+                  })
+        : ethers.constants.AddressZero; // native
```

Comment:
```suggestion
    let rlcInstanceAddress = isTokenMode
        ? getOrDeployRlc(deploymentOptions, owner) // token
        : ethers.constants.AddressZero; // native
```
with
```ts
async function getOrDeployRlc(deploymentOptions, owner) {
    return deploymentOptions.token
        ??  await new RLC__factory()
            .connect(owner)
            .deploy()
            .then((contract) => {
                contract.deployed();
                return contract.address;
            });
}
```
Or something similar. I'm just trying to make it more readable.

Reply:
Sure, see [Extract getOrDeployRlc](https://github.com/iExecBlockchainComputing/PoCo/pull/58/commits/106414b08ef5e9cbdc9a22913a63c5d30c1f1fe8) !